tests/extension/oneapi_weak_object/weak_object_common.h

```diff
@@ -0,0 +1,393 @@
+
+    bool result = !sycl::ext::oneapi::owner_less(w1, w2) &&
+                  (sycl::ext::oneapi::owner_less(w1, sycl_object) ||
+                   sycl::ext::oneapi::owner_less(w2, sycl_object));
+
+    INFO(
+        "Verify that owner_less compares equivalent for two weak objects that "
+        "are both empty");
+    CHECK(result);
+  }
+
+  void test_owner_before_different_objects(SYCLObjT sycl_object1,
+                                           SYCLObjT sycl_object2) {
+    auto w1 = sycl::ext::oneapi::weak_object<SYCLObjT>(sycl_object1);
+    auto w2 = sycl::ext::oneapi::weak_object<SYCLObjT>(sycl_object2);
+    {
+      bool weak_object_result = (w1.owner_before(w2) && !w2.owner_before(w1)) ||
+                                (!w1.owner_before(w2) && w2.owner_before(w1));
+
+      INFO(
+          "Verify that owner_before has some order for two weak object that "
+          "refer to different underlying SYCL objects");
+      CHECK(weak_object_result);
+    }
+    {
+      bool sycl_object_result = (sycl_object1.ext_aoneapi_owner_before(w2) &&
+                                 !sycl_object2.ext_aoneapi_owner_before(w1)) ||
+                                (!sycl_object1.ext_aoneapi_owner_before(w2) &&
+                                 sycl_object2.ext_aoneapi_owner_before(w1));
+
+      INFO(
+          "Verify that ext_aoneapi_owner_before has some order for two weak "
+          "object that refer to different underlying SYCL objects");
+      CHECK(sycl_object_result);
+    }
+    {
+      bool weak_object_referred_result =
+          w1.owner_before(sycl_object2) == w1.owner_before(w2) &&
+          w2.owner_before(sycl_object1) == w2.owner_before(w1);
+
+      INFO(
+          "Check that owner_before returns same values with weak object and "
+          "underlying SYCL object");
+      CHECK(weak_object_referred_result);
+    }
+    {
+      bool sycl_object_referred_result =
+          sycl_object1.ext_oneapi_owner_before(sycl_object2) ==
+              sycl_object1.ext_oneapi_owner_before(w2) &&
+          sycl_object2.ext_oneapi_owner_before(sycl_object1) ==
+              sycl_object2.ext_oneapi_owner_before(w1);
+
+      INFO(
+          "Check that ext_oneapi_owner_before returns same values with weak "
+          "object and underlying SYCL object");
+      CHECK(sycl_object_referred_result);
+    }
+  }
+
+  void test_owner_less_different_objects(SYCLObjT sycl_object1,
+                                         SYCLObjT sycl_object2) {
+    auto w1 = sycl::ext::oneapi::weak_object<SYCLObjT>(sycl_object1);
+    auto w2 = sycl::ext::oneapi::weak_object<SYCLObjT>(sycl_object2);
+    {
+      bool weak_object_result = (sycl::ext::oneapi::owner_less(w1, w2) &&
+                                 !sycl::ext::oneapi::owner_less(w2, w1)) ||
+                                (!sycl::ext::oneapi::owner_less(w1, w2) &&
+                                 sycl::ext::oneapi::owner_less(w2, w1));
```

Comment:
`owner_less` is a templated struct, so this should be changed to `sycl::ext::oneapi::owner_less<SYCLObjT>{}`. You could also reuse the same object for each comparison if that simplifies it.

Reply:
Applied.